flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGateTest.java

```diff
@@ -415,16 +395,19 @@ public void testRequestBuffersWithRemoteInputChannel() throws Exception {
 		final NetworkEnvironment network = createNetworkEnvironment();
 
 		try {
-			final ResultPartitionID resultPartitionId = new ResultPartitionID();
-			final ConnectionID connectionId = new ConnectionID(new InetSocketAddress("localhost", 5000), 0);
-			addRemoteInputChannel(network, inputGate, connectionId, resultPartitionId, 0);
+			RemoteInputChannel remoteFromUnknown =
+				InputChannelBuilder.newBuilder()
+					.setupFromNetworkEnvironment(network)
+					.buildUnknown(inputGate)
+					.toRemoteInputChannel(InputChannelBuilder.STUB_CONNECTION_ID);
```

Comment:
why not build remote directly?

Reply:
I tried to preserve the previous way of building remote but looking at it more, true, it does not seem to be relevant.